### PR TITLE
Change concatenation to use string templates instead of +

### PR DESCRIPTION
**Pull request by aaomidi**
Controversial change but we can take it if we want it.

### CODE REVIEW

**Comment by chlafreniere** [comment]
Are there perf implications here out of curiosity either way?

**Comment by aaomidi** [comment]
@chlafreniere It's mostly that + can cause unintended side effects since JS doesn't have types it can end up doing something completely unexpected. It's mostly for certainty of behavior.

**Comment by anthonydresser** [comment]
Most of the time templates are preferred to + in js since you know you get a string when you use templates, when using + it depends on the types you give it. Since we use typescript, typescript resolves those types for us, so its basically impossible (except a bug in typescript) that you will have confused types when using +.

This seems a bit unnecessary, but we also don't do that much string manipulation (since we use localize for everything visible) so its not like it will be a constant development blocker.

File: src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 				let updatedProfile = this._connectionManagementService.getConnectionProfileById(profile.id);
 				connectedContext = { connectionProfile: new ConnectionProfile(this._capabilitiesService, updatedProfile).toIConnectionProfile() };
 			} catch (err) {
-				this.logService.warn('Failed to connect due to error' + getErrorMessage(err));
+				this.logService.warn(`Failed to connect due to error${getErrorMessage(err)}`);
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts:142]: not introduced by you, but we should add a space or ': ' after 'due to error'

File: src/sql/workbench/contrib/jobManagement/browser/alertsView.component.ts [hunk 2]
```diff
@@ -215,10 +215,10 @@ export class AlertsViewComponent extends JobManagementView implements OnInit, On
...
 			'alertview-alertnameindicatordisabled';
 
-		return '<table class="alertview-alertnametable"><tr class="alertview-alertnamerow">' +
-			'<td nowrap class=' + resultIndicatorClass + '></td>' +
-			'<td nowrap class="alertview-alertnametext">' + dataContext.name + '</td>' +
-			'</tr></table>';
+		return `${'<table class="alertview-alertnametable"><tr class="alertview-alertnamerow">' +
+			'<td nowrap class='}${resultIndicatorClass}></td>` +
+			`<td nowrap class="alertview-alertnametext">${dataContext.name}</td>` +
+			`</tr></table>`;
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/jobManagement/browser/alertsView.component.ts:221]: for better readability, I'd say we should completely get rid of the single quotes and + symbols, I struggled to tell whether the changes look good. 😄  template string will preserve the line breaks.

File: src/sql/workbench/contrib/jobManagement/browser/jobsView.component.ts [hunk 3]
```diff
@@ -498,10 +498,10 @@ export class JobsViewComponent extends JobManagementView implements OnInit, OnDe
 				break;
 		}
 
-		return '<table class="jobview-jobnametable"><tr class="jobview-jobnamerow">' +
-			'<td nowrap class=' + resultIndicatorClass + '></td>' +
-			'<td nowrap class="jobview-jobnametext">' + escape(dataContext.name) + '</td>' +
-			'</tr></table>';
+		return `${'<table class="jobview-jobnametable"><tr class="jobview-jobnamerow">' +
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/jobManagement/browser/jobsView.component.ts:501]: same here, lets fully embrace template string

File: src/sql/workbench/contrib/jobManagement/browser/notebooksView.component.ts [hunk 4]
```diff
@@ -547,10 +547,10 @@ export class NotebooksViewComponent extends JobManagementView implements OnInit,
 				break;
 		}
 
-		return '<table class="jobview-jobnametable"><tr class="jobview-jobnamerow">' +
-			'<td nowrap class=' + resultIndicatorClass + '></td>' +
-			'<td nowrap class="jobview-jobnametext">' + escape(dataContext.name) + '</td>' +
-			'</tr></table>';
+		return `${'<table class="jobview-jobnametable"><tr class="jobview-jobnamerow">' +
```
Comment (alanrenmsft) [inline src/sql/workbench/contrib/jobManagement/browser/notebooksView.component.ts:550]: same here

**Review by alanrenmsft — APPROVED** [review]
took me a while to review this, hehe.
It would be nice if you can update the multiline strings i commented to fully use template string.

**Comment by aaomidi** [comment]
So I guess we should bring this up in a future meeting and see if we want to take this in. We could also just do it here, if anyone has any objections feel free to voice them!!

**Comment by aaomidi** [comment]
Ignore this

**Comment by aaomidi** [comment]
ignore this too

**Comment by aaomidi** [comment]
test

**Comment by aaomidi** [comment]
test

**Comment by aaomidi** [comment]
test

**Comment by aaomidi** [comment]
test

**Comment by aaomidi** [comment]
test

**Review by kburtram — APPROVED** [review]

**Comment by kburtram** [comment]
let's close this PR since it is a lot of code churn and isn't directly addressing any bugs, etc.

**Pull request closed without merging**